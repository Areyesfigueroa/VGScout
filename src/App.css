.App {
  margin:0;
  padding: 0;
}

html {
  background-color: rgba(0, 0, 0, 1);
}

.button {
  width: auto;
  height: auto;
  background-color: transparent;
  border: 0;
  text-align: center;
  color: #4a87ff;
  padding: 15px 10px;
  font-size: 100%;
  box-sizing: border-box;
  outline: none;
}

.button:hover {
  cursor: pointer;
}

.text {
  color: white;
  text-align: left;
  text-decoration: none;
}

/* VENDOR */
#tsparticles {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: -1;
}