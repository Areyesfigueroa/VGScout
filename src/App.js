import React, { Component } from 'react';
import './App.css';

import SearchBarContainer from "./containers/SearchBarContainers/SearchBarContainer";
import RedditPostsContainer from './containers/RedditPostsContainer/RedditPostsContainer';

import SectionWrapper from "./components/SectionWrapper/SectionWrapper";
import Background from "./components/Background/Background";
import ShadowBoxGradient from "./components/ShadowBoxGradient/ShadowBoxGradient";
import YoutubePlayer from "./components/YoutubePlayer/YoutubePlayer";
import Ratings from "./components/Ratings/Ratings";
import GameInfo from "./components/GameInfo/GameInfo";

//TESTING
import RedditPostsHolder from "./components/RedditPosts/RedditPostsMenu/RedditPostsMenu";

//TESTING
import { loadGameDetails, loadRedditPosts } from "./utils";

class App extends Component {

  state = {
    selectedValue: null,
    redditPostsData: null
  }

  clips = null;

  handleSelectedValue = (newValue) => {
    loadGameDetails(newValue.id).then(gameData => {
      loadRedditPosts(gameData.id).then(postData => {
        console.log(gameData);
        console.log(postData);
        this.setState({ 
          selectedValue: gameData,
          redditPostsData: postData
        });
      });
    });
  }

  render() {

    //Default Values
    let bgImage = require("./assets/background-default.png");
    let youtubePlayer = null;
    let ratings = null;
    let gameInfo = null;

    const parsePlatformData = () => {
      let platforms = this.state.selectedValue.platforms ? this.state.selectedValue.platforms: null;
      return platforms ? platforms.map(platformObj => platformObj.platform) : null;
    }

    //Once we have the game data loaded.
    if(this.state.selectedValue !== null) {
      let platformData = parsePlatformData();
      bgImage = this.state.selectedValue.background_image;

      youtubePlayer = (
        <YoutubePlayer 
        videoId={this.state.selectedValue.clip ? this.state.selectedValue.clip.video:null}
        />
      );

      //TODO, Find replacements for IGN and gamespot.
      ratings = (
        <Ratings
        metacritic={this.state.selectedValue.metacritic}
        ign={8}
        gamespot={4.5}
         />
      );

      gameInfo = (
        <GameInfo
        summary={this.state.selectedValue.description}
        developers={this.state.selectedValue.developers}
        publishers={this.state.selectedValue.publishers}
        platforms={platformData}
        genres={this.state.selectedValue.genres}
        />
      );
    }

    return (
      <div className="App">
        <ShadowBoxGradient degree={0} fromOpacity={0.2} toOpacity={0.9}/>
        <SectionWrapper>
          <SearchBarContainer 
          selectedValue={this.handleSelectedValue} 
          suggestionCount={3} 
          searchDelay={0.2} />
        </SectionWrapper>

        {/* If there is a selection */}
        <Background image={bgImage}/>
        <SectionWrapper>
          {youtubePlayer}
          {ratings}
          {gameInfo}
           {/* Reddit Comments */}
           <RedditPostsContainer data={this.state.redditPostsData} />
        </SectionWrapper>
      </div>
    );
  }
}

export default App;
