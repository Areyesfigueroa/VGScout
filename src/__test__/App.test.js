import React from 'react';
import App from '../App';

//Snapshots
import renderer from 'react-test-renderer';

//React's UI testing library
import {render,cleanup} from '@testing-library/react';
import {toBeInTheDocument} from '@testing-library/jest-dom';

expect.extend({toBeInTheDocument});
afterEach(cleanup);

describe("App Component", () => {
  it("Application can render without crashing", () => {
    const div = document.createElement("div");
    render(<App />, div);
  });

})