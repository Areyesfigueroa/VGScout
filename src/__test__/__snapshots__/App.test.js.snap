// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Match Snapshot 1`] = `
<div
  className="App"
>
  <section
    className="SectionWrapper"
    data-testid="sectionWrapper"
  >
    <div
      onKeyDown={[Function]}
    >
      <div
        className="SearchBar"
        data-testid="searchBar"
      >
        <input
          className="defaultSearch"
          id="search"
          onChange={[Function]}
          placeholder="Search..."
          type="search"
        />
        <button
          className="searchButton"
          onClick={[Function]}
        >
          <ion-icon
            name="search-outline"
          />
        </button>
        <button
          className="cancelButton"
          onClick={[Function]}
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <ion-icon
            name="close-circle-outline"
          />
        </button>
      </div>
    </div>
  </section>
</div>
`;
